### ISSUE TITLE
Language

### ISSUE THREAD

**Issue by Jojo-A** [MaxKravt/WiFi-Modbus-TCPtoRTU#1]
Hi MaxKravt,

I am really interested in your project. I think I was looking for exactly the same, as I have a ModbusRTU slave device (SDM630 Energy meter) and I like to integrate it into my home automation system "Home Assistant" over Wifi. So I was looking for a solution, how to covert ModbusTCP to ModbusRTU on the ESP8266 platform, which does nothing else but the conversion forth and back.

Sadly I do not speak any cyrilic language and can not read it.

Could you please tell me, if your project is what I need? Can I contribute somehow?

Thanks a lot in advance!

**Comment by MaxKravt** [comment]
Good day. 

The project is fully working. 
You can add a setting via telnet to a serial port and login password in configure.

Thanks.

**Comment by Jojo-A** [comment]
Hi and thank you for your reply!

I will definitely give it a try, as your project is one of the most promising I have found so far!

I've also made a fork, where I replaced all cyrillic text by a "raw" english Google translation. That made many things more clear ;) .

Thanks a lot for your effort!

**Comment by Jojo-A** [comment]
Hi,

thanks to you and your effort, I was able to integrate my smart-energy-meter SDM630 into my Home Automation System!
The setup is basically very simple: "Home Assistant" home automation server with ModbusTCP client (master), ESP8266 with your project (+minor modifications), low-cost TTL-RS485 transceiver module and the SDM630.

I implemented some minor changes to your project:
- translation of all cyrillic text to English
- replacement of the telnet configuration by a popular webserver configuration (WiFi configuration is done through a web interface instead of telnet console)
- usage of software-serial for ModbusRTU communication, to keep the hardware-serial free for debugging 

I also wanted to replace the current RTU-stuff by an existing Arduino library, but this was just a too heavy task for the beginning ;) .

Again, **thanks a ton** for your project!

**Comment by NickBuilder** [comment]
Hi!
I would really like to get this working so that I can communicate with my heating and ventilation systems.
I've managed to get my Wemos D1 mini to go online after configuring it using the telnet interface. However I dont have a clue on how to connect my rs-485 module to the Wemos.

As there is no serial pins specified in the code I guess the RX/TX pins are to be used but dont I also need a third pin handling the direction logic?

I have one of these rs-485 modules: https://www.aliexpress.com/i/1718630205.html
And a D1 mini clone: https://www.aliexpress.com/item/32813996238.html 

I've also looked at the fork by @Jojo-A as its mentioned in this conversation that a software-serial interface is used but I cant find a software-serial defined in the fork code.

Any hints on how to connect the rs-485 module would be much appreciated.
